### ISSUE TITLE
Amélioration : pouvoir faire une vente d'un composé à stock zéro V3.68.1 (1389baf V3.68)

### ISSUE THREAD

**Issue by patsymill** [Laurux/Laurux#102]
Pouvoir faire une vente d'un produit composé à stock zéro à partir du moment ou le (ou les) composants sont en stock.
Pourquoi : j'ai un client qui veut un article composé. Puisque je ne peux pas savoir à l'avance à quel composé mon composant va appartenir, je ne peux pas faire la facture immédiatement. Il faut d'abord que je fasse l'entrée en stock de mon composé (Stock / Entrée des produits composés). Je fais donc attendre ma clientèle !

**Comment by damscot** [comment]
je suis pas sur de bien comprendre comment tu utilises les produits
composés, je pense qu'a l'origine @Laurux avait fait ca pour des
assemblages de produits informatiques dans lequel l'assemblage devait etre
constitué avant vente
mais je sais pas trop... il est possible que si tu gères ta composition
comme un article non stocké tu puisse la vendre sans que la gestion de
stock te pose pb.

**Comment by PatrickMag** [comment]
La saisie des articles avec un stock négatif n'est pas obligatoirement bloquant, ça peut se faire. Par contre je suis assez d'accord, si le produit n'est pas composé par avance c'est des articles normaux que l'on vend.

**Comment by patsymill** [comment]
Oui, j'ai désactivé le contrôle stock négatif en facturation, Je peux donc saisir des articles composés sans stock. Par contre, cela ne destocke pas l'article composant, cela met le stock composé à -1. Il faudra quand même faire les entrées en stock des produits composés. 
Je vais donc faire un autre ticket pour pouvoir éditer les articles à stock négatif.

**Comment by damscot** [comment]
je prefere que l'on conserve ce ticket pour la gestion des stocks sur les articles composé...
je pense que cela risque de faire pas mal de travail @Laurux tu en pense quoi...
on a déjà pas mal de soucis avec la gestion de stock je ne voudrais pas que cette feature empire la situation.
2 eme question comment sont vus les articles composé en caisse ?

**Comment by lebounavo** [comment]
Bonjour,

Petite clarification sur une gestion de stock possible (je ne connais pas Laurux mais plutôt bien d'autres logiciels du secteur non libre) :

2 modes de gestion des composants sont possibles.
Gestion en "entrée" : 
- l'entrée d'un composé en stock génère des mouvements de sortie de stock des composants -> décrémente le stock des composants (gestion actuelle Laurux à ce que j'ai pu lire)
- la sortie de stock d'un composé est sans incidence sur les composants.
Gestion en "sortie" : 
- l'entrée d'un composé en stock est sans incidence sur les composants
- la sortie de stock d'un composé génère des mouvements de sortie de stock des composants -> décrémente le stock des composants.

La gestion en entrée est une gestion de fabrication de produit finis.
Attention pour un gestion rigoureuse des stocks, il est sage d'avoir des produits en-cours de façon à ce que le stock informatique reflète bien la réalité physique. Ainsi, il est préconisé de définir une nomenclature d'en-cours avec la logique suivante :

Nomenclature 1 : "Vélo (en cours fab)"
                    - Guidon
                    - Cadre
                    - Roue

Nomenclature 2 :  "Vélo (produit finis)"
                     - "Vélo (en cours fab)"

En début de fabrication, le stock "Vélo (en cours fab)" est augmenté sans que ce soit le stock de "Vélo (produit finis)" -non disponibles au demeurant- qui le soit.
Lorsque le vélo est terminé, le stock "Vélo (en cours fab)" est diminué du fait de l'entrée en stock du produit "Vélo (produit finis)".

La gestion en "sortie" est i.e. gestion d'assemblage à la livraison (mais pas que) est correspond peut-être à ce que Patsymill souhaite.

Un exemple concret de gestion combinée rencontré dans ma vie professionnelle : l'activité est celle d'un traiteur chez qui la gestion suivante a été mise en place avec succès (POC validé).

Nomenclature 1 "Menu/Plat" gérée en entrée :  "Menu 1"
                                                - Salade Carottes râpées/Céleris
                                                - Couscous

Nomenclature 2 "Plat/ingrédient" gérée en sortie :  "Salade Carottes râpées/Céleris"
                                                       - Carottes
                                                       - Céleris

Nomenclature 3 "Plat/ingrédient" gérée en sortie :  "Couscous"
                                                       - Semoule
                                                       - Merguez
                                                       - Boeuf
                                                       - Poulet
                                                       - Navets

L'entrée en fabrication décrémentait tous les ingrédients des plats concernés. Ajoutons que le logiciel utilisé permettait de modifier la composition d'une nomenclature directement dans le mouvement d'entrée en stock (sans modification de la fiche nomenclature/composé. Modification possible sur paramétrage -> notion de nomenclature variable ou fixe).

Le suivi des quantités en stock (i.e. la gestion de stock) pouvait être activé ou non au niveau du produit composé comme composant. La gestion de stock n'est peut-être pas utile pour le composé pour le cas de Patsymill (?).

En complément : La fiche nomenclature/composé autorisait le multi-niveau de composants avec choix d'édition ou non des lignes de composants (à n'importe quel niveau).

Puisse ces informations vous être utiles.
Désolé de ne pouvoir faire mieux i.e. de contribuer techniquement à votre projet ;-)
